includes/marketplace-suggestions/class-wc-marketplace-suggestions-api.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * REST API Marketplace suggestions API
+ *
+ * Handles requests for marketplace suggestions data
+ *
+ * @package WooCommerce/API
+ * @since   3.6.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Pull suggestion data from remote endpoint & cache in a transient.
+ *
+ * @return array of json API data
+ */
+function wc_marketplace_suggestions_get_api_data() {
+	$suggestion_data_url = 'https://d3t0oesq8995hv.cloudfront.net/add-ons/marketplace-suggestions.json';
+
+	$suggestion_data = get_transient( 'wc_marketplace_suggestions' );
+
+	if ( false !== $suggestion_data ) {
+		return $suggestion_data;
+	}
+
+	$raw_suggestions = wp_safe_remote_get(
+		$suggestion_data_url,
+		array( 'user-agent' => 'WooCommerce Marketplace Suggestions' )
+	);
+	if ( is_wp_error( $raw_suggestions ) ) {
+		return array();
+	}
+
+	// Parse the data to check for any errors.
+	// If it's valid, store structure in transient.
+	$suggestions = json_decode( wp_remote_retrieve_body( $raw_suggestions ) );
+	if ( $suggestions && is_array( $suggestions ) ) {
+		set_transient( 'wc_marketplace_suggestions', $suggestions, WEEK_IN_SECONDS );
+		return $suggestions;
+	}
+
+	return array();
+}
+
+/**
+ * Suggestion data GET handler.
+ */
+function wc_marketplace_suggestions_ajax_handler() {
+	$suggestion_data = wc_marketplace_suggestions_get_api_data();
+
+	echo wp_json_encode( $suggestion_data );
+
+	wp_die();
+}
+
+
+
+/**
+ * Return an array of suggestions the user has dismissed.
+ */
+function wc_marketplace_suggestions_get_dismissed() {
+	$dismissed_suggestions = array();
+
+	$dismissed_suggestions_data = get_user_option( 'wc_marketplace_suggestions_dismissed_suggestions', get_current_user_id() );
+	if ( $dismissed_suggestions_data ) {
+		$dismissed_suggestions = json_decode( $dismissed_suggestions_data );
+		if ( ! is_array( $dismissed_suggestions ) ) {
+			$dismissed_suggestions = array();
+		}
+	}
+
+	return $dismissed_suggestions;
+}
+
+/**
+ * Suggestion data GET handler.
+ */
+function wc_marketplace_suggestions_dismiss_handler() {
+	if ( ! check_ajax_referer( 'add_dismissed_marketplace_suggestion' ) ) {
+		wp_die();
+	}
+
+	$post_data       = wp_unslash( $_POST );
+	$suggestion_slug = sanitize_text_field( $post_data['slug'] );
```

Comment:
`wc_clean`

Reply:
Seems unnecessary, [just a wrapper for `sanitize_text_field`](https://docs.woocommerce.com/wc-apidocs/source-function-wc_clean.html#369-382). Happy to change this if that's preferred around here.